evalml/pipelines/utils.py

```diff
@@ -820,8 +826,21 @@ def _make_component_list_from_actions(actions):
     components = []
     cols_to_drop = []
     indices_to_drop = []
+
     for action in actions:
-        if action.action_code == DataCheckActionCode.DROP_COL:
+        if action.action_code == DataCheckActionCode.REGULARIZE_AND_IMPUTE_DATASET:
+            metadata = action.metadata
+            parameters = metadata.get("parameters", {})
+            components.extend(
+                [
+                    TimeSeriesRegularizer(
+                        time_index=parameters["time_index"],
```

Comment:
There's a check in `make_pipeline_from_data_check_output` if a time series problem type was passed and the problem configuration is invalid, which I feel like would cover this case. But I think if a user selects time series data checks individually and then passes in a non time series problem type, it could cause an issue. I'll err on the side of caution and add it, thanks!

Reply:
If this is being returned from `DateTimeFormatDataCheck` and the action code was `REGULARIZE_AND_IMPUTE_DATASET`, wouldn't parameters be passed in by default through the data check error?

Reply:
But true if we're potentially offering an empty dict above we should cover that case below as well, I'll add it because why not